### PR TITLE
Better handling of vehicle input limits and zeroing

### PR DESCRIPTION
**Pull request by adamdbrw**
Please also test whether the component parameters work as expected.

### CODE REVIEW

File: Code/Source/VehicleDynamics/VehicleInputsState.h [hunk 1]
```diff
@@ -7,12 +7,60 @@
...
+            return m_input;
+        }
+
+    private:
+        int64_t GetTimeSinceStartupUs() const
+        {
+            return static_cast<int64_t>(AZ::Interface<AZ::ITime>::Get()->GetElapsedTimeUs());
+        }
+
+        int64_t m_timeoutUs;
```
Comment (michalpelka) [inline Code/Source/VehicleDynamics/VehicleInputsState.h:55]:
```suggestion
     const int64_t m_timeoutUs;
```
I also do not understand the need for SFINAE in this class.

File: Code/Source/VehicleDynamics/VehicleModelLimits.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                        &VehicleModelLimits::m_steeringLimitRads,
+                        "Speed Limit",
+                        "Max steering angle (rads)");
+            }
+        }
+    }
+
+    float VehicleModelLimits::LimitValue(float value, float absoluteLimit)
+    {
+        return value >= 0 ? AZStd::min(value, absoluteLimit) : AZStd::max(value, -absoluteLimit);
```
Comment (michalpelka) [inline Code/Source/VehicleDynamics/VehicleModelLimits.cpp:43]: Consider omitting branching:
```suggestion
return AZStd::min(AZStd::max(value, -absoluteLimit), absoluteLimit) ;
```

**Review by michalpelka — COMMENTED** [review]
Minor things from my side. 
I did not run it yet.

File: Code/Source/VehicleDynamics/VehicleModelLimits.cpp (hunk 2 above)
Comment (adamdbrw) [reply Code/Source/VehicleDynamics/VehicleModelLimits.cpp:43]: While the second option works and is shorter, I believe it is a bit less intuitive to understand.

File: Code/Source/VehicleDynamics/VehicleInputsState.h (hunk 1 above)
Comment (adamdbrw) [reply Code/Source/VehicleDynamics/VehicleInputsState.h:55]: Primary reason for including it is that it informs the developer of the intent for the templated type without having him look into implementation. 
Also, compilation errors for invalid type are different, but I guess it is up to personal preference which is easier to understand.

Comment (adamdbrw) [reply Code/Source/VehicleDynamics/VehicleInputsState.h:55]: Another issue is that some non-arithmetic types can have zero assigned, which is not something we would like (e.g. std::string).

File: Code/Source/VehicleDynamics/VehicleModelLimits.cpp (hunk 2 above)
Comment (adamdbrw) [reply Code/Source/VehicleDynamics/VehicleModelLimits.cpp:43]: Replaced with AZStd::clamp, which is there :)

**Pull request merged**